textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -0,0 +1,142 @@
+
+package edu.uci.ics.textdb.dataflow.dictionarymatcher;
+
+import java.util.List;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IDictionary;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.IntegerField;
+import edu.uci.ics.textdb.common.field.StringField;
+
+/**
+ * @author Sudeep [inkudo]
+ * 
+ */
+public class DictionaryMatcher implements IOperator {
+
+    private IOperator operator;
+    private IDictionary dict;
+    private String dictValue;
+    private int positionIndex; // next position in the field to be checked.
+    private int fieldIndex; // Index of the next field to be checked.
+    private int spanIndexVal; // Starting position of the matched dictionary
+                              // string
+    private ITuple spanTuple;
+    private List<IField> fields;
+
+    public DictionaryMatcher(IDictionary dict, IOperator operator) {
+        this.operator = operator;
+        this.dict = dict;
+    }
+
+    /**
+     * @about Opens dictionary matcher. Initializes positionIndex and fieldIndex
+     *        Gets first sourceoperator tuple and dictionary value.
+     */
+    @Override
+    public void open() throws Exception {
+        try {
+            positionIndex = 0;
+            fieldIndex = 0;
+            operator.open();
+            dictValue = dict.getNextDictValue();
+            spanTuple = operator.getNextTuple();
+            fields = spanTuple.getFields();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @about Gets next matched tuple. Returns a new span tuple including the
+     *        span results. Performs a scan based search, gets the dictionary
+     *        value and scans all the documents for matches
+     */
+    @Override
+    public ITuple getNextTuple() throws Exception {
+        if (fieldIndex < fields.size()) {
+            IField field = spanTuple.getField(fieldIndex);
```

Comment:
could you name the field in a better way that we can tell which one is from dictionary side or the input side?